packages/core/src/client.ts

```diff
@@ -205,36 +206,13 @@ export class Client extends AsyncEventEmitter<ManagerShardEventsMap> {
 	 * @param options - The options for the request
 	 * @param timeout - The timeout for waiting for each guild members chunk event
 	 */
-	public async requestGuildMembers(options: GatewayRequestGuildMembersData, timeout = 10_000) {
+	public async *requestGuildMembersIterator(options: GatewayRequestGuildMembersData, timeout = 10_000) {
 		const shardId = calculateShardId(options.guild_id, await this.gateway.getShardCount());
 		const nonce = options.nonce ?? DiscordSnowflake.generate().toString();
 
-		const promise = new Promise<RequestGuildMembersResult>((resolve, reject) => {
-			const members: RequestGuildMembersResult['members'] = [];
-			const notFound: RequestGuildMembersResult['notFound'] = [];
-			const presences: RequestGuildMembersResult['presences'] = [];
-
-			const timer = setTimeout(() => {
-				reject(new Error('Request timed out'));
-			}, timeout);
-
-			const handler = ({ data }: MappedEvents[GatewayDispatchEvents.GuildMembersChunk][0]) => {
-				timer.refresh();
-
-				if (data.nonce !== nonce) return;
-
-				members.push(...data.members);
-				if ('presences' in data) presences.push(...data.presences);
-				if ('not_found' in data) notFound.push(...data.not_found);
-
-				if (data.chunk_index >= data.chunk_count - 1) {
-					this.off(GatewayDispatchEvents.GuildMembersChunk, handler);
-					resolve({ members, nonce, notFound, presences });
-				}
-			};
-
-			this.on(GatewayDispatchEvents.GuildMembersChunk, handler);
-		});
+		const timer = setTimeout(() => {
+			throw new Error('Request timed out');
+		}, timeout);
```

Comment:
This won't work by the way, you need to use an `AbortController` and pass its `signal` to [`events.on`](https://nodejs.org/dist/latest-v20.x/docs/api/events.html#eventsonemitter-eventname-options)'s options.